### DISCUSSION TITLE
Is there a way to pre-validate the configuration CSV file before deploying?

### DISCUSSION THREAD

**Discussion by maturmel** [Azure-Player/azure.datafactory.tools#218] [Q&A] [answered]
I am in a heavier development phase and there are a lot of changes occurring in our Data Factory, sometimes the devs forget to adjust the config CSV accordingly (ex.: deleting/renaming a pipeline and not updating the name in CSV). Is there a command I can run to validate the CSV against the Data Factory to check references are all good? Right now my only way of knowing is through Azure DevOps' release pipeline stage to QA, which is too "late" in the process, and also only gives "the next error" instead of all of them. There's a lot of deployment cycles wasted on fixing config CSV.

Thanks for any ideas on how I can improve my process!

**Comment by NowinskiK** [comment, marked as answer]
Yes, there is something exactly for this reason:
`Test-AdfCode -RootFolder "$RootFolder"`

**Reply by maturmel** [reply]
Thank you! It seems to do the trick but the output is only showing it validated my config-prod.csv and not my config-qa.csv. Is this to be expected or should it validate all deployment CSVs ?

**Reply by NowinskiK** [reply]
It should validate all files in `deployment` folder

**Reply by maturmel** [reply]
Yes it did the second time around. I probably misread the log the first time :)

Cheers
